src/NodeServices/PythonServices.cs

```diff
@@ -0,0 +1,272 @@
+﻿using Dynamo.Utilities;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Collections.Specialized;
+using System.Linq;
+using System.Reflection;
+
+namespace Dynamo.PythonServices
+{
+    /// <summary>
+    /// Enum of possible values of python engine versions.
+    /// TODO: Remove when dynamic loading logic is there then we no longer need a hard copy of the options
+    /// </summary>
+    public enum PythonEngineVersion
+    {
+        Unspecified,
+        IronPython2,
+        CPython3
+    }
+
+    /// <summary>
+    /// Singleton class that other class can access and use for query loaded Python Engine info.
+    /// TODO: Make the Singleton class public when Dynamic loading is also ready.
+    /// </summary>
+    public sealed class PythonEngineManager
```

Comment:
this class is/will be the old PythonEngineSelector (which will be removed)

Reply:
whats the reasoning for making this public now?

Reply:
I will be needed for the new API (PythonEngineManager.Instance.AvailableEngines).